contracts/DistributionCreator.sol

```diff
@@ -204,118 +218,107 @@ contract DistributionCreator is UUPSHelper, ReentrancyGuardUpgradeable {
-                successfullFrom[count] = froms[i];
-                campaignReallocation[_campaignId][froms[i]] = to;
-                campaignListReallocation[_campaignId].push(froms[i]);
-                count++;
+        for (uint256 i; i < fromsLength; ) {
+            campaignReallocation[_campaignId][froms[i]] = to;
+            campaignListReallocation[_campaignId].push(froms[i]);
+            unchecked {
+                ++i;
             }
         }
-        assembly {
-            mstore(successfullFrom, count)
-        }
-
-        if (count == 0) revert Errors.InvalidOverride();
-        emit CampaignReallocation(_campaignId, successfullFrom, to);
+        emit CampaignReallocation(_campaignId, froms, to);
+    }
+
+    /// @notice Increases the predeposited token balance of a `user` for a given `rewardToken`
+    /// @dev If a governor is calling the function, the user must have sent the tokens beforehand
+    /// @dev This function can be used to deposit on behalf of another user
+    /// @dev This function MUST NOT be used to deposit a rebasing token
+    function increaseTokenBalance(address user, address rewardToken, uint256 amount) external {
+        if (!accessControlManager.isGovernor(msg.sender))
+            IERC20(rewardToken).safeTransferFrom(msg.sender, address(this), amount);
+        _updateBalance(user, rewardToken, creatorBalance[user][rewardToken] + amount);
+    }
+
+    /// @notice Decreases the predeposited token balance of a `user` for a given `rewardToken`
+    /// @dev Only the user themselves or a governor can call this function
+    function decreaseTokenBalance(
+        address user,
+        address rewardToken,
+        address to,
+        uint256 amount
+    ) external onlyUserOrGovernor(user) {
+        _updateBalance(user, rewardToken, creatorBalance[user][rewardToken] - amount);
+        IERC20(rewardToken).safeTransfer(to, amount);
+    }
+
+    /// @notice Increases the token allowance of an `operator` for a `user`
+    function increaseTokenAllowance(
+        address user,
+        address operator,
+        address rewardToken,
+        uint256 amount
+    ) external onlyUserOrGovernor(user) {
+        _updateAllowance(user, operator, rewardToken, creatorAllowance[user][operator][rewardToken] + amount);
+    }
+
+    /// @notice Decreases the token allowance of an `operator` for a `user`
+    function decreaseTokenAllowance(
+        address user,
+        address operator,
+        address rewardToken,
+        uint256 amount
+    ) external onlyUserOrGovernor(user) {
+        _updateAllowance(user, operator, rewardToken, creatorAllowance[user][operator][rewardToken] - amount);
+    }
+
+    /// @notice Toggles the ability of an `operator` to manage campaigns on behalf of a `user`
+    /// @dev Only the user themselves or a governor can call this function
+    function toggleCampaignOperator(address user, address operator) external onlyUserOrGovernor(user) {
```

Comment:
why not just having the mapping 'creatorAllowance' and if allowance non null --> valid operator

Reply:
for the sake of composability and having different addresses possible, like most people may not split roles, but seems cleaner to have different roles